src/Microsoft.VisualStudio.Threading/TplExtensions.cs

```diff
@@ -108,6 +108,128 @@ public static async Task<T> WithTimeout<T>(this Task<T> task, TimeSpan timeout)
             return task.GetAwaiter().GetResult();
         }
 
+        /// <summary>
+        /// Creates a new <see cref="CancellationToken"/> that is canceled when any of a set of other tokens are canceled.
+        /// </summary>
+        /// <param name="original">The first token.</param>
+        /// <param name="other">The second token.</param>
+        /// <returns>A struct that contains the combined <see cref="CancellationToken"/> and a means to release memory when you're done using it.</returns>
+        public static CombinedCancellationToken CombineWith(this CancellationToken original, CancellationToken other)
+        {
+            if (original.IsCancellationRequested)
```

Comment:
💭 The alternative here is:

```csharp
if (original.IsCancellationRequested || !other.CanBeCanceled)
```

Not sure if the pair of these is simpler than the ternary at the end, but it seems to make the method "appear" less complex.

Reply:
Sure, I can go with that.